src/main/java/com/iotiq/user/internal/UserRepository.java

```diff
@@ -14,4 +15,8 @@ public interface UserRepository extends JpaRepository<User, UUID>, JpaSpecificat
 
     boolean existsByAccountInfoUsername(String username);
 
+    boolean existsByPersonalInfoEmail(String email);
+
+    @Query("SELECT COUNT(u) FROM User u WHERE u.personalInfo.email = :email AND u.id <> :id")
+    int countUsersWithPersonalInfoEmailAndNotId(UUID id, String email);
```

Comment:
I think you could have also used the `exists` keyword here and refrain from writing a query.

Reply:
Noted!